Testing/UnitTests/xASL_ut_UnitTest_function_ExploreASL_Master.m

```diff
@@ -0,0 +1,269 @@
+if ~isfield(x,'ProcessArray'),  testCondition = false;      end
+if ~isfield(x,'SkipPause'),     testCondition = false;      end
+if ~isfield(x,'iWorker'),       testCondition = false;      end
+if ~isfield(x,'nWorkers'),      testCondition = false;      end
+
+% Now let's check the values
+if isfield(x,'DataParPath')
+    if ~isempty(x.DataParPath) || ~ischar(x.DataParPath)
+        testCondition = false;
+    end
+end
+if isfield(x,'ImportArray')
+    if length(x.ImportArray)<4 || sum(x.ImportArray)>0 || ~isnumeric(x.ImportArray)
+        testCondition = false;
+    end
+end
+if isfield(x,'ProcessArray')
+    if length(x.ProcessArray)<3 || sum(x.ProcessArray)>0 || ~isnumeric(x.ProcessArray)
+        testCondition = false;
+    end
+end
+if isfield(x,'SkipPause')
+    if length(x.SkipPause)>1 || x.SkipPause~=1 || ~isnumeric(x.SkipPause)
+        testCondition = false;
+    end
+end
+if isfield(x,'iWorker')
+    if length(x.iWorker)>1 || x.iWorker~=1 || ~isnumeric(x.iWorker)
+        testCondition = false;
+    end
+end
+if isfield(x,'nWorkers')
+    if length(x.nWorkers)>1 || x.nWorkers~=1 || ~isnumeric(x.nWorkers)
+        testCondition = false;
+    end
+end
+
+% Get test duration
+UnitTest.tests(3).duration = toc(testTime);
+
+% Evaluate your test
+UnitTest.tests(3).passed = testCondition;
+
+
+%% Test run 4
+
+% Give your individual subtest a name
+UnitTest.tests(4).testname = 'Initialize (DCM2NIFTI)';
+
+% Start the test
+testTime = tic;
+
+% Copy test patient
+testPatientSource = fullfile(TestRepository,'UnitTesting','synthetic_dcm','test_patient'); 
+testPatientDestination = fullfile(TestRepository,'UnitTesting','working_directory','test_patient');
+xASL_Copy(testPatientSource, testPatientDestination);
+
+% Read test files
+[x] = ExploreASL_Master(fullfile(testPatientDestination,'sourceStructure.json'),[1 0 0 0],0,1,1,1);
+
+% Define one or multiple test conditions here
+testCondition = true; % Fallback
+if ~exist(fullfile(testPatientDestination,'analysis'),'dir'),                   testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1'),'dir'),            testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1','ASL_1'),'dir'),    testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1','T1w_1'),'dir'),    testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1','ASL_1','ASL4D.json'),'file'),      testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1','ASL_1','ASL4D.nii'),'file'),       testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1','ASL_1','M0.json'),'file'),         testCondition = false;          end
+if ~exist(fullfile(testPatientDestination,'analysis','Sub1','ASL_1','M0.nii'),'file'),          testCondition = false;          end
```

Comment:
I try to use as few ExploreASL functions within the UnitTesting as possible. Otherwise you test functions with functions you need to test. Basically nothing wrong with that, if you test xASL_exist first, but I try to get rid of these scenarios whenever possible.
In addition, I don't want it to find both .nii & .nii.gz here. We want the test to be strict. We always want the same results, not sometimes .nii and sometimes .nii.gz, right?